scaleway/resource_ipam_ip_test.go

```diff
@@ -0,0 +1,282 @@
+
+		l.Debugf("sweeper: deleting the IPs in (%s)", region)
+
+		listIPs, err := ipamAPI.ListIPs(&ipam.ListIPsRequest{Region: region}, scw.WithAllPages())
+		if err != nil {
+			return fmt.Errorf("error listing ips in (%s) in sweeper: %s", region, err)
+		}
+
+		for _, v := range listIPs.IPs {
+			err := ipamAPI.ReleaseIP(&ipam.ReleaseIPRequest{
+				IPID:   v.ID,
+				Region: region,
+			})
+			if err != nil {
+				l.Debugf("sweeper: error (%s)", err)
+
+				return fmt.Errorf("error releasing IP in sweeper: %s", err)
+			}
+		}
+
+		return nil
+	})
+}
+
+func TestAccScalewayIPAMIP_Basic(t *testing.T) {
+	tt := NewTestTools(t)
+	defer tt.Cleanup()
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:          func() { testAccPreCheck(t) },
+		ProviderFactories: tt.ProviderFactories,
+		CheckDestroy:      testAccCheckScalewayIPAMIPDestroy(tt),
+		Steps: []resource.TestStep{
+			{
+				Config: `
+					resource scaleway_vpc vpc01 {
+						name = "my vpc"
+					}
+
+					resource scaleway_vpc_private_network pn01 {
+						vpc_id = scaleway_vpc.vpc01.id
+						ipv4_subnet {
+							subnet = "172.16.32.0/22"
+						}
+					}
+
+					resource scaleway_ipam_ip ip01 {
+					 source {
+						private_network_id = scaleway_vpc_private_network.pn01.id
+					  }
+					}
+				`,
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckScalewayIPAMIPExists(tt, "scaleway_ipam_ip.ip01"),
+					resource.TestCheckResourceAttr("scaleway_ipam_ip.ip01", "is_ipv6", "false"),
+					testAccCheckScalewayResourceRawIDMatches("scaleway_ipam_ip.ip01", "source.0.private_network_id", "scaleway_vpc_private_network.pn01", "id"),
+					resource.TestCheckResourceAttrSet("scaleway_ipam_ip.ip01", "source.0.subnet_id"),
+					resource.TestCheckResourceAttrSet("scaleway_ipam_ip.ip01", "address"),
+					resource.TestCheckResourceAttrSet("scaleway_ipam_ip.ip01", "created_at"),
+					resource.TestCheckResourceAttrSet("scaleway_ipam_ip.ip01", "updated_at"),
+				),
+			},
+		},
+	})
+}
+
+func TestAccScalewayIPAMIP_WithStandaloneAddress(t *testing.T) {
+	tt := NewTestTools(t)
+	defer tt.Cleanup()
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:          func() { testAccPreCheck(t) },
+		ProviderFactories: tt.ProviderFactories,
+		CheckDestroy:      testAccCheckScalewayIPAMIPDestroy(tt),
+		Steps: []resource.TestStep{
+			{
+				Config: `
+					resource scaleway_vpc vpc01 {
+						name = "my vpc"
+					}
+
+					resource scaleway_vpc_private_network pn01 {
+						vpc_id = scaleway_vpc.vpc01.id
+						ipv4_subnet {
+							subnet = "172.16.32.0/22"
+						}
+					}
+
+					resource scaleway_ipam_ip ip01 {
+					 address = "172.16.32.7"
+					 source {
+						private_network_id = scaleway_vpc_private_network.pn01.id
+					  }
+					}
+				`,
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckScalewayIPAMIPExists(tt, "scaleway_ipam_ip.ip01"),
+					resource.TestCheckResourceAttr("scaleway_ipam_ip.ip01", "address", "172.16.32.7/22"),
```

Comment:
Is it normal that address is a IPv4 without CIDR and in the test is with a CIDR?

Reply:
Yes its normal, the API permits to the user to book an IP with CIDR and without it (in this case it needs to fall within the correct range of the allocatable addresses of the PN's subnet), the API always returns the IP with CIDR in the response.